ZenPacks/zenoss/OpenStackInfrastructure/datasources/HeartbeatsAMQPDataSource.py

```diff
@@ -97,106 +90,11 @@ def config_key(cls, datasource, context):
-                    heartbeat_hostname = possible_hostname
-                    break
-
-            process_heartbeats = {}
-
-            if heartbeat_hostname is None:
-                # We have not heard a heartbeat from this host under
-                # any of its possible hostnames (short name, fqdn)
-                #
-                # So there won't be any process heartbeats, and we will
-                # consider them all to be down.
-                process_heartbeats = {}
-            else:
-                process_heartbeats = last_heard_heartbeats[heartbeat_hostname]
-
-            for proc in required_processes:
-                if proc not in process_heartbeats:
-                    evt = {
-                        'device': device_id,
-                        'severity': ZenEventClasses.Warning,
-                        'eventKey': hostname + '_' + proc,
-                        'summary': "No heartbeats received from '%s' on %s.  Check the host and process status." % (proc, hostname),
-                        'eventClassKey': 'openStackCeilometerHeartbeat',
-                    }
-                    log.error(pformat(evt))
-                else:
-                    # diff > MAX_TIME_LAPSE?
-                    time_diff = time() - \
-                        process_heartbeats[proc]['lastheard']
-
-                    if time_diff > MAX_TIME_LAPSE:
-                        evt = {
-                            'device': device_id,
-                            'severity': ZenEventClasses.Warning,
-                            'eventKey': hostname + '_' + proc,
-                            'summary': "No heartbeats received from '%s' on %s for more than %d seconds.  Check its status and restart it if necessary." % (proc, hostname, MAX_TIME_LAPSE),
-                            'eventClassKey': 'openStackCeilometerHeartbeat',
-                        }
-                        log.error(pformat(evt))
-                    else:
-                        evt = {
-                            'device': device_id,
-                            'severity': ZenEventClasses.Clear,
-                            'eventKey': hostname + '_' + proc,
-                            'summary': "Process '%s' on %s is sending heartbeats normally." % (proc, hostname),
-                            'eventClassKey': 'openStackCeilometerHeartbeat',
-                        }
-
-                data['events'].append(evt)
-
-        if len(data['events']):
-            data['events'].append({
-                'device': device_id,
-                'summary': 'OpenStack Ceilometer AMQP Heartbeat: successful collection',
-                'severity': ZenEventClasses.Clear,
-                'eventKey': 'openstackCeilometerAMQPHeartbeatCollection',
-                'eventClassKey': 'openstack-Heartbeats',
-                })
-
-        defer.returnValue(data)
+        """This method is deprecated. Do not use!"""
+        return
```

Comment:
pretty sure you need to have a yield in here if you're going to use inlineCallbacks.  (otherwise your method is not a generator)    yield None is probably fine

Reply:
https://twistedmatrix.com/documents/13.2.0/api/twisted.internet.defer.inlineCallbacks.html indicates that a plain "return"  will cause the Deferred to have a result of None and will not result in error.

Reply:
Correct, but it still requires there to be one yield in there somewhere, to make it into a generator in the first place.

Reply:
I removed the Inlinecallback, which is allowed by theplugin.
Please re-review.

Reply:
It still needs to return a deferred though, wither or not you use inlinecallbacks to make that happen
   https://github.com/zenoss/ZenPacks.zenoss.PythonCollector/blob/develop/README.html#L259